content/articles/auto-typing-js/index.md

```diff
@@ -0,0 +1,279 @@
+
+The general syntax of an IIFE is shown below:
+
+```js
+(function () {
+  statements;
+
+})();
+
+```
+
+In the above example, the grouping operator prevents variables within the IIFE from being accessed from the global scope. 
+
+> You can learn more about IIFEs from official documentation found [here](https://developer.mozilla.org/en-US/docs/Glossary/IIFE). 
+
+```javascript
+function autoType(elementClass, typingSpeed, timeout) {
+    var ourClass = $(elementClass);
+    ourClass.css({
+      "position": "relative",
+      "display": "inline-block"
+    });
+    ourClass = ourClass.find(".text-js");
+    var text = ourClass.text().trim().split('');
+    var amntOfChars = text.length;
+    var newString = "";
+    setTimeout(function () {
+      ourClass.css("opacity", 1);
+      ourClass.text("");
+      for (var i = 0; i < amntOfChars; i++) {
+        (function (i, char) {
+          setTimeout(function () {
+            newString += char;
+            ourClass.text(newString);
+          }, i * typingSpeed);
+        })(i, text[i]);
+      }
+    }, timeout);
+  }
+
+function fire(){
+    autoType(".type-js", 85, 1000);
+    autoType(".type-js-ii", 200, 2500);  
+    autoType(".type-js-iii", 85, 5800);  
+    autoType(".type-js-iv", 200, 9300);   
+    autoType(".type-js-v", 85, 15000);    
+} 
+```
+
+The code consists of a function `autoType()` and an IIFE inside the function.
+
+`autoType()` contains three parameters:
+1. `elementClass` - the target class where the auto-typing will be invoked
+2. `typingSpeed` - This is the speed of typing the text out
+3. `timeout` - the timeout period for waiting =before the `autoTyping()` function is fired.
+
+WE then have a variable `ourClass` for storing the target class. The next line is where we set the display style of the class.
+
+Next, we look for `text-js`. This is a class found in our target class. It contains the text that will be typed out. Thereafter, we remove any trailing spaces.
+
+The `amntOfChars` will store the length of text.
+We will store the text to display inside the `newString` variable.
+
+Inside the `setTimeout()` function, we start by setting the opacity of the target class's contents to 1 since we initially set it to 0 as we will see later. We do set it to 0 because we don't want it to be visible at first sight. The text of the target class is temporarily cleared before display.
+
+
+Next is a `for-loop` where the IIFE is now used. The IIFE takes the index and the text character at the particular index as parameters. It then the timeout as the index number multiplied by the typing speed. The `newString`  is concatenated with the text character at the index until the characters of the sentence are finished.
+
+THe function `fire()` is used to call the `auto-typing()` function.
+
+#### HTML and CSS
+
+Below is the HTML code:
```

Comment:
Tell the reader to create a html file in a particular location before jumping straight to the code.